library/network_connections.py

```diff
@@ -74,7 +74,8 @@ def check_output(argv, lang = None):
             ev = os.environ.copy()
             ev['LANG'] = lang if lang is not None else 'C'
             p = subprocess.Popen(argv, stdout=subprocess.PIPE, stderr=DEVNULL, env=ev)
-            out = p.communicate()[0]
+            # FIXME: Can we assume this to always be UTF-8?
+            out = p.communicate()[0].decode("UTF-8")
             if p.returncode != 0:
```

Comment:
I suppose we should assume it is the current locale, if it is an output of a command? That's the point of a locale after all.

Reply:
That said in another locale we will not be able to parse the output. Could we omit `lang` and set `LC_ALL` to `C` unconditionally and then parse the output as pure ASCII?

Reply:
Maybe `Util.check_output()` should just return whatever python gives us. The caller must know how to interpret this and `decode()` it accordingly.

Reply:
> I doubt that many people use it to set some other encoding than UTF-8

I think we should not rely on such assumptions.

> The locale is usually about translation

That was my point. If one translates `ethtool` output, it will not be possible to parse it. Given that,
> I plan to drop support for the lang parameter, too, because we will not want to use some other locale here in the module.

makes perfect sense.

> I guess hardcoding to C and decoding as ascii should work unless we really approach something that is not ascii

One should hardcode C to ensure no translation is used if one relies on that.

> unless we really approach something that is not ascii

We could revisit it when such a use case appears. By the way,

> `cat /path/to/utf-8-file.txt` will not necessarily be correct ascii.

`cat /path/to/binary-file.txt`will not be correct UTF-8 either... I think the correct answer is "don't do that"...